app/registration.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Register components (via a list of glob patterns)
+ *
+ * Copyright © 2015 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NonComposerComponentRegistration;
```

Comment:
Taking into account another @davidalger 's suggestion with anonymous function, maybe we can use anonymous function for the whole registration logic as well? So we should not care about names at all.